### PR TITLE
Add announcement CZI EOSS6

### PR DESCRIPTION
**Pull request by glemaitre**
Adding the announcement for the CZI EOSS 6 award

### CODE REVIEW

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+title: "Scikit-learn awarded CZI EOSS Cycle 6 Grant"
+date: August 6, 2024
+categories:
+  - Funding
+tags:
+  - Open Source
+  - Sponsor
```
Comment (reshamas) [inline _posts/2024-08-06-czi-eoss6-announcement.md:8]:
```suggestion
  - Funding
```

Comment (glemaitre) [reply _posts/2024-08-06-czi-eoss6-announcement.md]: Thanks.

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 2]
```diff
@@ -18,4 +18,75 @@ postauthors:
...
+use cases.
+
+On top of all these items, we intend to continue working on the general maintenance of
+the project, addressing bug reports and performance regressions. As a community-driven
+project, we want to dedicate time reviewing external contributions.
+
+## References
+
+[1] [Scikit-learn proposal for CZI EOSS
+6](https://github.com/scikit-learn/administrative/blob/master/czi_eoss_proposal/EOSS6-0000000551_202312181059.pdf)
```
Comment (reshamas) [inline _posts/2024-08-06-czi-eoss6-announcement.md:92]:
```suggestion
[1] [Scikit-learn proposal for CZI EOSS
6](https://github.com/scikit-learn/administrative/blob/master/czi_eoss_proposal/EOSS6-0000000551_202312181059.pdf)
[x] [Prior CZI EOSS grants](https://chanzuckerberg.com/science/programs-resources/open-science/grants/)
```

Comment (glemaitre) [reply _posts/2024-08-06-czi-eoss6-announcement.md:92]: Uhm I'm not interested in citing the full list of grants. What is the reasoning?

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+---
+title: "Scikit-learn awarded CZI EOSS Cycle 6 Grant"
```
[the remark this answers is not in the document]
Comment (glemaitre) [reply _posts/2024-08-06-czi-eoss6-announcement.md]: I took the second version: this is more than an extension since this is a new proposal :).

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  - name: Lucy Liu
+    website: https://github.com/lucyleeow
+    image: lucyliu.jpeg
+---
+<div>
+  <img src="/assets/images/posts_images/{{ page.featured-image }}" alt="">
+  {% include postauthor.html %}
+</div>
+
+We are delighted to announce that `scikit-learn` is awarded a grant from
```
Comment (betatim) [inline _posts/2024-08-06-czi-eoss6-announcement.md:26]:
```suggestion
We are delighted to announce that `scikit-learn` has been awarded a grant from
```
or maybe "was awarded"?

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+the [Chan Zuckerberg Initiative (CZI)](https://chanzuckerberg.com/)'s [Essential Open
+Source Software for Science
+(EOSS)](https://chanzuckerberg.com/rfa/essential-open-source-software-for-science/)
+program. This grant is funded by [Wellcome Trust](https://wellcome.org/).
+As in previous round, this cycle supports open-source software projects that are
```
Comment (betatim) [inline _posts/2024-08-06-czi-eoss6-announcement.md:31]:
```suggestion
As in previous rounds, this cycle supports open-source software projects that are
```

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+essential to biomedical research. This is the third time that CZI EOSS supports
+`scikit-learn`. In previous rounds, the following projects were funded:
+
+- [CZI EOSS 1](https://chanzuckerberg.com/eoss/proposals/scikit-learn-maintenance-and-enhancement-for-gradient-boosting/)
+  helped at creating to the
```
Comment (betatim) [inline _posts/2024-08-06-czi-eoss6-announcement.md:36]:
```suggestion
  helped create the
```

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  [`HistGradientBoostingClassifier`](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html) and
+  [`HistGradientBoostingRegressor`](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingRegressor.html) estimators.
+  These estimators are the equivalent of gradient boosting models implemented in
```
Comment (betatim) [inline _posts/2024-08-06-czi-eoss6-announcement.md:39]:
```suggestion
  These estimators are equivalent to the gradient boosting models implemented in
```

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+  `LightGBM` and `XGBoost`.
+- [CZI EOSS 4](https://chanzuckerberg.com/eoss/proposals/maintenance-extension-of-scikit-learn-machine-learning-in-python/)
+  extended `scikit-learn` to work better with missing values and categorical data in
```
Comment (betatim) [inline _posts/2024-08-06-czi-eoss6-announcement.md:42]:
```suggestion
  extended `scikit-learn` to improve handling of missing values and categorical data in
```

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+  several estimators.
+
+Both grants allowed us to maintain and enhance `scikit-learn` to better serve the
+community.
+
+In this new grant, we will focus on improving the evaluation and inspection of
```
Comment (betatim) [inline _posts/2024-08-06-czi-eoss6-announcement.md:48]: Should we move this further up? It seems like telling the reader what this grant is about is the most important thing in this article?

But also, telling people about the amazing hist gradient boosting is important :-/

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+predictive models. We provide in the next section a detailed description of the
+proposed work.
```
Comment (betatim) [inline _posts/2024-08-06-czi-eoss6-announcement.md:50]:
```suggestion
predictive models. In the next section we will describe the
proposed work.
```

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+
+## Predictive models evaluation & inspection
+
+When building a machine learning pipeline for a specific research problem, two key
+aspects are closely connected: (i) design the pipeline and (ii) assess, analyze, and
+inspect it. Researchers strive to identify the optimal pipeline, maximizing specific
+evaluation metrics, while also seeking at explaining the validity and rationale behind
+the pipeline's predictions. This is the corner stone to properly answering research
```
Comment (betatim) [inline _posts/2024-08-06-czi-eoss6-announcement.md:58]:
```suggestion
the pipeline's predictions. This is the cornerstone of answering research
```

I think it is "cornerstone of", if we want to say "... to" we need something like "key to". I'd stick with cornerstone. Removed "properly" because it didn't read great somehow and I couldn't think of a way to shuffle things around

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+questions. With this proposal we aim to improve and extend the available `scikit-learn`
+tools.
+
+`scikit-learn` provides building blocks for model evaluation and statistical analysis of
+results. Originally, this information was presented in a raw format and required
+expertise from scientists to create intuitive supports for outreach to peers and
```
Comment (betatim) [inline _posts/2024-08-06-czi-eoss6-announcement.md:64]:
```suggestion
expertise from scientists to create intuitive reports for outreach to peers and
```
Guessing this should be "report"? Pretty sure "supports" isn't the right word here.

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+display pipelines in these environments. However, there is room for improvement in terms
+of user interaction and accessibility. Finally, as `scikit-learn` is widely used as a
+reference package, it is crucial to improve the section of the library dedicated to
+model explainability. We aim to improve the documentation and user experience with the
+existing explainability tools, making sure that they use the appropriate tool for their
+use cases.
+
+On top of all these items, we intend to continue working on the general maintenance of
+the project, addressing bug reports and performance regressions. As a community-driven
+project, we want to dedicate time reviewing external contributions.\
```
Comment (betatim) [inline _posts/2024-08-06-czi-eoss6-announcement.md:90]:
```suggestion
project, we want to dedicate time reviewing external contributions.
```

**Review by betatim — COMMENTED** [review]
Nice work! Left a few suggestions that I thought of while reading it.

File: _posts/2024-08-06-czi-eoss6-announcement.md (hunk 9 above)
Comment (glemaitre) [reply _posts/2024-08-06-czi-eoss6-announcement.md:48]: Yep, I had also some doubts about the structure. Let me move the past grants in a section at the end of the post.

File: _posts/2024-08-06-czi-eoss6-announcement.md (hunk 2 above)
Comment (reshamas) [reply _posts/2024-08-06-czi-eoss6-announcement.md]: It looks like the URL doesn't include the filtering for scikit-learn, so the individual grant links will do.

File: _posts/2024-08-06-czi-eoss6-announcement.md [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+`scikit-learn` provides building blocks for model evaluation and statistical analysis of
+results. Originally, this information was presented in a raw format and required
+expertise from scientists to create intuitive reports for outreach to peers and
+outsiders. Recently, the `scikit-learn` community developed displays to easily generate
+visual figures for communicating such results. However, these displays are still in
+their early development stages and do not leverage all available statistical analysis
+tools (i.e., cross-validation) from `scikit-learn`. Thus, we aim to expand these
+displays to use the right statistical tools and thus promote the adoption of best
+practices when reporting results. Additionally, we also intend to create new displays
+that are considered as "standards" during such analyses.
```
Comment (lucyleeow) [inline _posts/2024-08-06-czi-eoss6-announcement.md:57]: Are you saying that the new displays will be 'standard' (i.e., commonly used/well-established in the field) or that using the new displays will become standard practice in people's analysis workflows?

**Review by lucyleeow — APPROVED** [review]
Looks great, thanks!

I may be confused, but is SLEP021 work included in this grant?

File: _posts/2024-08-06-czi-eoss6-announcement.md (hunk 2 above)
Comment (glemaitre) [reply _posts/2024-08-06-czi-eoss6-announcement.md]: I linked each proposal to the CZI pages instead of having references.

File: _posts/2024-08-06-czi-eoss6-announcement.md (hunk 14 above)
Comment (glemaitre) [reply _posts/2024-08-06-czi-eoss6-announcement.md:58]: Let me reformulate it but I wanted to say that there is "standard" analysis and we are going to implement displays to depict visually those analysis.

Comment (lucyleeow) [reply _posts/2024-08-06-czi-eoss6-announcement.md]: Perfect, it looks great!

**Review by ogrisel — APPROVED** [review]
LGTM as well!

**Comment by glemaitre** [comment]
Reading again, I see that the SLEP is not mentioned. So I added a line on this particular work and the ultimate goal. I think the blog post looks good. I'll merge now and I'll start to make the announcement on the different social media.

**Comment by glemaitre** [comment]
Thanks everyone.
